File: src/B_9012/B_9012.java
package B_9012;

import java.io.BufferedWriter;
import java.io.IOException;
import java.io.OutputStreamWriter;
import java.util.Scanner;
public class B_9012 {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Scanner sc = new Scanner(System.in);
        int t=sc.nextInt();
        for(int i=0; i<t; i++){
            String n=sc.next();
            String res = vps(n);
            bw.write(res+"\n");
            bw.flush();
        }
    }
    public static String vps(String n){
        int count=0;
        for(int i=0; i<n.length(); i++){
            if(n.charAt(i)=='(') count++;
            else if(count==0&&n.charAt(i)==')') return "NO";
            else count--;
        }
        if(count==0) return "YES";
        else return "NO";
    }
}

File: src/B_9012/B_9012.kt
package B_9012

import java.io.BufferedWriter
import java.io.IOException
import java.io.OutputStreamWriter
import java.util.*



@Throws(IOException::class)
fun main(args: Array<String>) {
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val sc = Scanner(System.`in`)
    val t = sc.nextInt()
    for (i in 0 until t) {
        val n = sc.next()
        val res = vps(n)
        bw.write(
            """
                  $res
                  
                  """.trimIndent()
        )
        bw.flush()
    }
}

fun vps(n: String): String {
    var count = 0
    for (i in 0 until n.length) {
        if (n[i] == '(') count++ else if (count == 0 && n[i] == ')') return "NO" else count--
    }
    return if (count == 0) "YES" else "NO"
}
